src/Core/Manifest/ModuleManifest.php

```diff
@@ -225,4 +238,31 @@ public function getModules()
     {
         return $this->modules;
     }
+
+    /**
+     * Get modules sorted by priority
+     *
+     * @return Module[]
+     */
+    public function getSortedModules()
+    {
+        if (!$this->sortedModules) {
+            $order = static::config()->uninherited('module_priority');
+            $project = static::config()->get('project');
+
+            $sorter = Injector::inst()->createWithArgs(
+                PrioritySorter::class.'.modulesorter',
+                [
+                    $this->modules,
+                    $order
+                ]
+            )
+            ->setVariable(self::PROJECT_KEY, $project)
+            ->setDefaultTop(self::PROJECT_KEY);
```

Comment:
"top" should be controlled by config. See my earlier comment on ordering of $project. I don't think there should be a get/setDefaultTop at all.

I suggest to break out the YML config for ordering of modules as below:

file `framework/_config/modules.yml`

```yaml
---
name: modulepriorityproject
before: *
---
SilverStripe\Core\Manifest\ModuleManifest
  module_priority:
    - $project
---
name: modulepriorityother
---
SilverStripe\Core\Manifest\ModuleManifest
  module_priority:
    - $other_modules
```

We should probably use `$other_modules` for consistency with the `$projects` placeholder, since they are both placeholders.